src/prefect/orion/api/block_schemas.py

```diff
@@ -40,16 +42,18 @@ async def create_block_schema(
             detail="Block schemas for protected block types cannot be created.",
         )
 
+    block_schema_checksum = Block._calculate_schema_checksum(block_schema.fields)
+    existing_block_schema = await models.block_schemas.read_block_schema_by_checksum(
+        session=session, checksum=block_schema_checksum
+    )
+    if existing_block_schema:
+        response.status_code = status.HTTP_200_OK
+        return existing_block_schema
```

Comment:
does the client do anything with this response, or does it not matter because the checksums are the same?

Reply:
The client doesn't currently differentiate between a 200 or a 201. This response is primary used to know which block schema to associate a block document with.